### PR TITLE
fix(date-dropdown): updated the current date when the props are changed

### PR DESCRIPTION
**Pull request by ridvandmrc**
## Pull request checklist

Please check if your PR fulfills the following requirements:

- [X] Tests for the changes have been added (for bug fixes / features)
- [X] Docs have been reviewed and added / updated if needed (for bug fixes / features)
- [X] Build (`yarn build`) was run locally and any changes were pushed
- [X] Unit tests (`yarn test`) were run locally and passed
- [X] Visual Regression Tests (`yarn visual-regression`) were run locally and passed
- [X] Linting (`npm lint`) was run locally and passed

## Pull request type

Please check the type of change your PR introduces:

- [X] Bug fix
- [ ] Feature
- [ ] Refactoring (no functional changes, no API changes)
- [ ] Build related changes
- [ ] Documentation content changes
- [ ] Other (please describe):

## What is the current behavior?
date-dropdown can not be set outside of the component

GitHub Issue Number: #1122 

## What is the new behavior?
Fixed and tested

-
-
-

## Does this introduce a breaking change?

- [ ] Yes
- [X] No

### CODE REVIEW

**Review by nuke-ellington — CHANGES_REQUESTED** [review]
Looks like this breaks a unit test. Also a test case for the change was not added.

**Comment by ridvandmrc** [comment]
Hi @nuke-ellington ,
I tested it to much but I can not reproduced

**Comment by ridvandmrc** [comment]
> > Hi @nuke-ellington , I tested it to much but I can not reproduced
> 
> @ridvandmrc Do you plan to update this PR? Regarding @nuke-ellington change request?
> 
> There are also some debug console log statements inside the component code, you should remove the logs there.

Yes, I plan to update, and I remove every logs

**Comment by ridvandmrc** [comment]
Hi @danielleroux ,
I updated the test cases and also I added an example for react-sample and also tested manually

**Review by danielleroux — CHANGES_REQUESTED** [review]

File: packages/react-test-app/src/preview-examples/date-dropdown.tsx
[the remark this answers is not in the document]
Comment (ridvandmrc) [reply packages/react-test-app/src/preview-examples/date-dropdown.tsx:1]: Hi @danielleroux,
I was thinking that these examples could be more helpful. 
okay I revert it, you can review again.

[the remark this answers is not in the document]
Comment (danielleroux) [reply packages/react-test-app/src/preview-examples/date-dropdown.tsx:1]: Yes they are helpful but you have missed also to adapt the angular, vue and html examples they have to be all the same.

And you have to keep in mind that the examples as to fit into the example iframe window also on different screen sizes.

[the remark this answers is not in the document]
Comment (ridvandmrc) [reply packages/react-test-app/src/preview-examples/date-dropdown.tsx:1]: I see thanks a lot

**Review by danielleroux — DISMISSED** [review]

**Review by danielleroux — APPROVED** [review]

File: .changeset/orange-insects-pull.md [hunk 1]
```diff
@@ -0,0 +1,5 @@
+---
+"@siemens/ix": patch
+---
+
+fix(core/date-dropdown): update the date when the props are change
```
Comment (nuke-ellington) [inline .changeset/orange-insects-pull.md:5]:
```suggestion
fix(core/date-dropdown): update the date when the props are changed
```

**Review by nuke-ellington — CHANGES_REQUESTED** [review]

**Review by nuke-ellington — APPROVED** [review]

**Pull request merged**